src/vec.rs

```diff
@@ -38,32 +38,42 @@ impl LateLintPass for UselessVec {
             let TypeVariants::TyRef(_, ref ty) = cx.tcx.expr_ty_adjusted(expr).sty,
             let TypeVariants::TySlice(..) = ty.ty.sty,
             let ExprAddrOf(_, ref addressee) = expr.node,
-            let Some(vec_args) = unexpand_vec(cx, addressee)
         ], {
-            let snippet = match vec_args {
-                VecArgs::Repeat(elem, len) => {
-                    format!("&[{}; {}]", snippet(cx, elem.span, "elem"), snippet(cx, len.span, "len")).into()
-                }
-                VecArgs::Vec(args) => {
-                    if let Some(last) = args.iter().last() {
-                        let span = Span {
-                            lo: args[0].span.lo,
-                            hi: last.span.hi,
-                            expn_id: args[0].span.expn_id,
-                        };
+            check_vec_macro(cx, expr, addressee);
+        }}
+
+        // search for `for _ in vec![…]`
+        if let Some((_, arg, _)) = recover_for_loop(expr) {
+            check_vec_macro(cx, arg, arg);
+        }
+    }
+}
+
+fn check_vec_macro(cx: &LateContext, expr: &Expr, vec: &Expr) {
+    if let Some(vec_args) = unexpand_vec(cx, vec) {
+        let snippet = match vec_args {
+            VecArgs::Repeat(elem, len) => {
+                format!("&[{}; {}]", snippet(cx, elem.span, "elem"), snippet(cx, len.span, "len")).into()
+            }
+            VecArgs::Vec(args) => {
+                if let Some(last) = args.iter().last() {
+                    let span = Span {
+                        lo: args[0].span.lo,
+                        hi: last.span.hi,
+                        expn_id: args[0].span.expn_id,
+                    };
 
-                        format!("&[{}]", snippet(cx, span, "..")).into()
-                    }
-                    else {
-                        "&[]".into()
-                    }
+                    format!("&[{}]", snippet(cx, span, "..")).into()
+                }
+                else {
+                    "&[]".into()
                 }
-            };
+            }
+        };
 
-            span_lint_and_then(cx, USELESS_VEC, expr.span, "useless use of `vec!`", |db| {
-                db.span_suggestion(expr.span, "you can use a slice directly", snippet);
-            });
-        }}
+        span_lint_and_then(cx, USELESS_VEC, expr.span, "useless use of `vec!`", |db| {
+            db.span_suggestion(expr.span, "you can use a slice directly", snippet);
+        });
     }
```

Comment:
That’s for that kind of ugly diff that I’d want the GitHub diff view to have a `--ignore-all-space` option.